### PR TITLE
go/libraries/doltcore/sqle: Refactored Auto Increment

### PR DESCRIPTION
**Pull request by andrew-wm-arthur**
companion [PR](https://github.com/dolthub/go-mysql-server/pull/886)

### CODE REVIEW

File: go/libraries/doltcore/sqle/globalstate/auto_increment_tracker.go [hunk 1]
```diff
@@ -15,146 +15,117 @@
...
+	"github.com/dolthub/dolt/go/libraries/doltcore/schema"
+)
 
-	// Case 0: Just use the value passed in.
-	potential, ok := a.valuePerTable[tableName]
-	if !ok {
-		// Use the disk val if the table has not been initialized yet.
-		potential = diskVal
+// CoerceAutoIncrementValue converts |val| into an AUTO_INCREMENT sequence value
+func CoerceAutoIncrementValue(val interface{}) (seq uint64, err error) {
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/globalstate/auto_increment_tracker.go:30]: Don't name the return params if you don't use them

File: go/libraries/doltcore/sqle/globalstate/auto_increment_tracker.go [hunk 2]
```diff
@@ -15,146 +15,117 @@
...
-	}
+// Set sets the current AUTO_INCREMENT value for |tableName|.
+func (a AutoIncrementTracker) Set(tableName string, val uint64) {
+	a.mu.Lock()
+	defer a.mu.Unlock()
+	a.sequences[tableName] = val
+}
+
+// AddTable adds |tablename| to the AutoIncrementTracker.
+func (a AutoIncrementTracker) AddTable(tableName string) {
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/globalstate/auto_increment_tracker.go:120]: Prefer NewTable here

File: go/libraries/doltcore/sqle/writer/noms_table_writer.go [hunk 3]
```diff
@@ -56,14 +55,15 @@ type nomsTableWriter struct {
...
-	autoIncCol  schema.Column
 	vrw         types.ValueReadWriter
 	kvToSQLRow  *index.KVToSqlRowConverter
 	tableEditor editor.TableEditor
 	sess        WriteSession
-	aiTracker   globalstate.AutoIncrementTracker
 	batched     bool
 
+	autoInc globalstate.AutoIncrementTracker
+	autoOrd int
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/writer/noms_table_writer.go:65]: What is this if there is no auto inc column? -1? Document this

File: go/libraries/doltcore/sqle/writer/sessioned_table_editor.go [hunk 4]
```diff
@@ -123,15 +123,8 @@ func (ste *sessionedTableEditor) HasEdits() bool {
...
-// GetAutoIncrementValue implements TableEditor.
-func (ste *sessionedTableEditor) GetAutoIncrementValue() types.Value {
-	return ste.tableEditor.GetAutoIncrementValue()
-}
-
-// SetAutoIncrementValue implements TableEditor.
-func (ste *sessionedTableEditor) SetAutoIncrementValue(v types.Value) error {
-	ste.dirty = true
-	return ste.tableEditor.SetAutoIncrementValue(v)
+func (ste *sessionedTableEditor) SetDirty(dirty bool) {
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/writer/sessioned_table_editor.go:126]: Method comment plz

Comment (zachmu) [reply go/libraries/doltcore/sqle/writer/sessioned_table_editor.go:126]: Also this is only ever called with true, would change to `MarkDirty()`

**Review by zachmu — APPROVED** [review]
LGTM, good refactoring

File: go/libraries/doltcore/sqle/writer/noms_table_writer.go (hunk 3 above)
Comment (andrew-wm-arthur) [reply go/libraries/doltcore/sqle/writer/noms_table_writer.go:65]: it was unused, deleted it

**Pull request merged**